server/src/test/scala/za/co/absa/atum/server/api/database/flows/functions/GetFlowCheckpointsIntegrationTests.scala

```diff
@@ -0,0 +1,42 @@
+package za.co.absa.atum.server.api.database.flows.functions
+
+import za.co.absa.atum.server.ConfigProviderTest
+import za.co.absa.atum.model.dto.{CheckpointQueryDTO, PartitionDTO, PartitioningDTO}
+import za.co.absa.atum.server.api.TestTransactorProvider
+import za.co.absa.atum.server.api.database.PostgresDatabaseProvider
+import zio.interop.catz.asyncInstance
+import zio.test.Assertion.failsWithA
+import zio.{Scope, ZIO}
+import zio.test._
+
+object GetFlowCheckpointsIntegrationTests extends ConfigProviderTest {
+
+  override def spec: Spec[TestEnvironment with Scope, Any] = {
+
+    val partitioningDTO1: PartitioningDTO = Seq(
+      PartitionDTO("stringA", "stringA"),
+      PartitionDTO("stringB", "stringB")
+    )
+
+    suite("GetFlowCheckpointsIntegrationTests")(
+      test("Returns expected sequence of flow of Checkpoints with existing partitioning") {
+        val partitioningQueryDTO: CheckpointQueryDTO = CheckpointQueryDTO(
+          partitioning = partitioningDTO1,
+          limit = Some(10),
+          checkpointName = Some("checkpointName")
+        )
+
+        for {
+          getFlowCheckpoints <- ZIO.service[GetFlowCheckpoints]
+          exit <- getFlowCheckpoints(partitioningQueryDTO).exit
+        } yield assert(exit)(failsWithA[doobie.util.invariant.NonNullableColumnRead])
```

Comment:
The database call should never end up in NonNullableColumnRead exception. That signals nothing but incorrect implementation of the fa-db class. As of now all columns that are potentially returned with NULL value have to be modeled as Option.